go/master/client.go

```diff
@@ -20,6 +25,51 @@ type Client struct {
 	ch   chan []byte
 }
 
+// MasterAddresser provide master address
+type masterAddresser struct {
+	client    *clientv3.Client
+	timeout   time.Duration
+	endpoints []string
+}
+
+// Address return the address
+func (m masterAddresser) Address() string {
+	for {
+		ctx, cancel := context.WithTimeout(context.Background(), m.timeout)
+		resp, err := m.client.Get(ctx, masterAddrPath)
+		cancel()
+		if err != nil {
+			log.Errorf("Fetch master addr failed, reconnecting to etcd, %v", err)
+			err := m.client.Close()
+			if err != nil {
+				log.Errorln(err)
+				time.Sleep(m.timeout)
+				continue
+			}
+			// reconnect to etcd server
```

Comment:
It seem that etcd have a some kind of retry method for `grpc` but don't know how to use yet, may be help here https://github.com/coreos/etcd/blob/master/clientv3/retry.go#L28